PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:5 + add:0, change matches:
-    _inherit = 'mrp.production'
-    _inherit = 'mrp.workorder'
-    _inherit = "project.project"
-    _inherit = 'stock.move'
-    _inherit = 'stock.rule'

Total Changes: 12

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/project_mrp_account/models/mrp_production.py
+++ b/addons/project_mrp_account/models/mrp_production.py
@@ -4,7 +4,7 @@ from odoo import api, fields, models, _
 
 
 class MrpProduction(models.Model):
-    _inherit = 'mrp.production'
+    _inherit = ['mrp.production']
 
     has_analytic_account = fields.Boolean(compute='_compute_has_analytic_account')
 

--- a/addons/project_mrp_account/models/mrp_workorder.py
+++ b/addons/project_mrp_account/models/mrp_workorder.py
@@ -4,7 +4,7 @@ from odoo import models
 
 
 class MrpWorkorder(models.Model):
-    _inherit = 'mrp.workorder'
+    _inherit = ['mrp.workorder']
 
     def _create_or_update_analytic_entry_for_record(self, value, hours):
         super()._create_or_update_analytic_entry_for_record(value, hours)

--- a/addons/project_mrp_account/models/project_project.py
+++ b/addons/project_mrp_account/models/project_project.py
@@ -4,8 +4,8 @@ from odoo import models
 from odoo.osv import expression
 
 
-class Project(models.Model):
-    _inherit = "project.project"
+class ProjectProject(models.Model):
+    _inherit = ["project.project"]
 
     # ----------------------------
     #  Project Updates

--- a/addons/project_mrp_account/models/stock_move.py
+++ b/addons/project_mrp_account/models/stock_move.py
@@ -6,7 +6,7 @@ from odoo.tools import format_list
 
 
 class StockMove(models.Model):
-    _inherit = 'stock.move'
+    _inherit = ['stock.move']
 
     def _get_analytic_distribution(self):
         distribution = self.raw_material_production_id.project_id._get_analytic_distribution()

--- a/addons/project_mrp_account/models/stock_rule.py
+++ b/addons/project_mrp_account/models/stock_rule.py
@@ -4,7 +4,7 @@ from odoo import models
 
 
 class StockRule(models.Model):
-    _inherit = 'stock.rule'
+    _inherit = ['stock.rule']
 
     def _prepare_mo_vals(self, product_id, product_qty, product_uom, location_id, name, origin, company_id, values, bom):
         res = super()._prepare_mo_vals(product_id, product_qty, product_uom, location_id, name, origin, company_id, values, bom)
